### PR TITLE
fix lint + pre commit

### PR DESCRIPTION
**Pull request by fmoral2**
#### Types of Changes ####

https://github.com/rancher/distros-test-framework/pull/53#discussion_r1425158452 
 Check this comment to explanations 

- adding pre commit
- fix lint

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.
 - ok 

3. Update the documentation if needed.
- ok 

4. Update makefile and docker run if adding new tests.
 -ok 

5. Run your tests at least 4 times with all configurations needed and possible.
- ok

6. If needed test with different os types.

 -ok

### CODE REVIEW

File: docs/conventions.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 ### Testcase naming convention:
 - All tests should be placed under `./testcase/<TESTNAME>`.
 - All test functions should be named: `Test<TESTNAME>`.
+
+### Contributing:
+- All contributions should be made via pull requests.
+- Before pushing please run `make pre-commit` to format your code and run linters.
+- All pull requests should be made against the `main` branch.
+- All pull requests should be reviewed by at least two other person before being merged.
+- All pull requests should be squashed into a single commit before being merged.
```
Comment (rancher-max) [inline docs/conventions.md:16]: Can we change this one to: "All Pull Requests should use the Squash and Merge option when being merged"?

File: pkg/template/helper.go [hunk 2]
```diff
@@ -28,7 +28,9 @@ func updateExpectedValue(template TestTemplate) {
 	}
 }
 
-// executeTestCombination get a template and pass it to `processTestCombination` to execute test combination on group of IPs
+// executeTestCombination get a template and pass it to `processTestCombination`
+//
```
Comment (rancher-max) [inline pkg/template/helper.go:32]: Is it best practice to have an "empty new line" like this?

File: Makefile [hunk 3]
```diff
@@ -267,7 +267,14 @@ test-restart-service:
...
+#vet-lint:
+#	@echo "Running go vet and lint"
+#	@go vet ./${TEST_DIR} && golangci-lint run --tests
+
+.PHONY: pre-commit
+pre-commit:
+	@gofmt -s -w .
+	@goimports -w .
+	@go vet ./...
+	@golangci-lint run --tests ./...
```
Comment (rancher-max) [inline Makefile:280]: I'm also thinking maybe we just ensure the golangci-lint runs and blocks if this is working well -- that way we don't have to all have golangci-lint installed locally. But I'm fine either way

File: .golangci.yaml [hunk 4]
```diff
@@ -55,14 +52,17 @@ linters-settings:
...
         arguments: ["fmt.Printf", "builder.WriteString"]
       - name: unnecessary-stmt
       - name: unreachable-code
       - name: unused-parameter
       - name: unused-receiver
 
+run:
+  skip-files:
+    - "pkg/testcase/selinux.go"
+    -  ".*_suite_test.go"
```
[the remark this answers is not in the document]
Comment (fmoral2) [reply .golangci.yaml:65]: sorry i thought i had that on the PR description , will add there also

- suite_test files we have flag definitions , which honestly doesnt need or even makes sense break line for those, so on this file really makes sense to disable the lint, which is what has warnings for it 

so summary, this is far more readable
Screenshot 2023-12-13 at 07 30 20


than this =
> 
Screenshot 2023-12-13 at 07 30 48

[the remark this answers is not in the document]
Comment (fmoral2) [reply .golangci.yaml:65]: selinux file same situation on the context variables

File: Makefile [hunk 5]
```diff
@@ -267,7 +267,14 @@ test-restart-service:
...
 #========================= TestCode Static Quality Check =========================#
-.PHONY: vet-lint
-vet-lint:
-	@echo "Running go vet and lint"
-	@go vet ./${TEST_DIR} && golangci-lint run --tests
\ No newline at end of file
+#.PHONY: vet-lint
+#vet-lint:
+#	@echo "Running go vet and lint"
+#	@go vet ./${TEST_DIR} && golangci-lint run --tests
```
[the remark this answers is not in the document]
Comment (fmoral2) [reply Makefile:273]: absolutely

my bad

File: docs/conventions.md (hunk 1 above)
Comment (fmoral2) [reply docs/conventions.md:16]: sure

File: pkg/template/helper.go (hunk 2 above)
Comment (fmoral2) [reply pkg/template/helper.go:32]: for bigger doc comments yes, so they get formatted on documentation

File: Makefile (hunk 3 above)
Comment (fmoral2) [reply Makefile:280]: yes, for this u will have to have this and also imports installed 
but the idea of this pre commit is kind of that , so we all start to have the same local formatting , avoiding adding unnecessary warnings on lint CI also facilitating lives for reviewers that dont need to look for more basic stuff on code review

File: .golangci.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (rancher-max) [reply .golangci.yaml:65]: Makes sense, thank you!

**Review by ShylajaDevadiga — APPROVED** [review]

File: pkg/testcase/ingressdns.go [hunk 6]
```diff
@@ -83,8 +83,8 @@ func TestIngressRoute(applyWorkload, deleteWorkload bool, apiVersion string) {
 			Expect(err).NotTo(HaveOccurred(), "failed to read file for ingressroute resource")
 		}
 
-		publicIp := fmt.Sprintf("%s.nip.io", shared.FetchNodeExternalIP()[0])
-		replacer := strings.NewReplacer("$YOURDNS", publicIp, "$APIVERSION", apiVersion)
+		pIp := fmt.Sprintf("%s.nip.io", shared.FetchNodeExternalIP()[0])
```
Comment (aganesh-suse) [inline pkg/testcase/ingressdns.go:86]: NIT: curious.. why are we changing this var name to pIp from publicIp - publicIp feels like a better descriptive name?

Comment (fmoral2) [reply pkg/testcase/ingressdns.go:86]: hey sorry to merge that if you still had questions

so basically here https://github.com/fmoral2/distros-test-framework/blob/a2dddd8f85a09677493014bc0bf33ba272dad74f/pkg/testcase/ingressdns.go#L74

its already a var with the same name, to no create a shadow or we need to rename or create something similar to a closure , which in the context would be an overkill

**Pull request merged**